File: frontend/src/api/tips.ts
// 内核提示（tips）API。
//
// 提示是**内核能力**：文案存放在内核语言包的 `tips.items.<序号>` 命名空间下，
// 后端只负责挑选 key，前端拿到 key 后用 `t(key)` 渲染 —— 这样用户切换语言时，
// 正在显示的提示会立刻跟着换语言，无需重新请求后端。

import { call } from "./core";

/** 一条提示：`key` 用于 i18n 渲染，`text` 是后端按当前语言解析好的文案。 */
export interface TipView {
  key: string;
  text: string;
}

/** 取一条随机提示；当前语言下无可用提示（语言包缺失）时返回 `null`。 */
export function tipsNext(): Promise<TipView | null> {
  return call<TipView | null>("tips_next");
}

/** 当前语言下全部提示 key（按序号排序）。 */
export function tipsKeys(): Promise<string[]> {
  return call<string[]>("tips_keys");
}

File: src-tauri/src/commands/tips.rs
//! 提示命令：向所有前端（内核 Shell 与各模块）暴露提示能力。
//!
//! 设计意图：提示是**内核能力**，模块不自行维护提示文案，统一经此取用，
//! 保证全应用提示风格与语言一致。

use serde::Serialize;
use tauri::State;

use crate::error::CommandResult;
use crate::state::KernelContext;

/// 一条提示的视图。`key` 供前端 `t()` 渲染（语言切换即时生效），
/// `text` 为当前语言下的文案（便于非 i18n 场景直接展示）。
#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "snake_case")]
pub struct TipPayload {
    pub key: String,
    pub text: String,
}

/// 取一条随机提示；当前语言无可用提示时返回 `null`。
#[tauri::command]
pub fn tips_next(kernel: State<'_, KernelContext>) -> CommandResult<Option<TipPayload>> {
    Ok(kernel.tips().next().map(|t| TipPayload {
        key: t.key,
        text: t.text,
    }))
}

/// 当前语言下全部提示 key（按序号排序）。
#[tauri::command]
pub fn tips_keys(kernel: State<'_, KernelContext>) -> CommandResult<Vec<String>> {
    Ok(kernel.tips().keys())
}
